receiver/awsnvmereceiver/internal/instancestore/scraper.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package instancestore
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/aws/amazon-cloudwatch-agent/receiver/awsnvmereceiver/internal/metadata"
+	"github.com/aws/amazon-cloudwatch-agent/receiver/awsnvmereceiver/internal/nvme"
+)
+
+const InstanceStoreMagic uint32 = 0xEC2C0D7E
+
+var ErrInvalidInstanceStoreMagic = errors.New("invalid Instance Store magic number")
+
+// Metrics represents the parsed Instance Store metrics from the NVMe log page.
+type Metrics struct {
+	Magic                 uint32
+	Reserved              uint32
+	ReadOps               uint64
+	WriteOps              uint64
+	ReadBytes             uint64
+	WriteBytes            uint64
+	TotalReadTime         uint64
+	TotalWriteTime        uint64
+	EBSIOPSExceeded       uint64 // Not applicable
+	EBSThroughputExceeded uint64 // Not applicable
+	EC2IOPSExceeded       uint64
+	EC2ThroughputExceeded uint64
+	QueueLength           uint64
+	NumHistograms         uint32
+	NumBins               uint32
+	IOSizeRange           [8]uint32
+	Bounds                [32]struct {
+		Lower uint64
+		Upper uint64
+	}
+	Histograms   [5]HistogramPair
+	ReservedArea [888]byte
+}
+
+type HistogramPair struct {
+	Read  [32]uint64
+	Write [32]uint64
+}
+
+func (Metrics) IsNVMeMetrics() {}
+
+// Scraper implements DeviceTypeScraper for Instance Store devices.
+type Scraper struct{}
+
+func NewScraper() *Scraper {
+	return &Scraper{}
+}
+
+func (s *Scraper) Model() string {
+	return "Amazon EC2 NVMe Instance Storage"
+}
+
+func (s *Scraper) DeviceType() string {
+	return "instance_store"
+}
+
+func (s *Scraper) Identifier(serial string) (string, error) {
+	if strings.TrimSpace(serial) == "" {
+		return "", fmt.Errorf("empty serial for instance store")
+	}
+	return serial, nil
```

Comment:
Trimming the space is fine, but it looks like we don't actually return the modified value, so if the serial is "    serial", then we'll be sending that as is. Is that intentional?